diff --git a/src/base_bigstring_stubs.c b/src/base_bigstring_stubs.c
index 4bf39db..8f4905a 100644
--- a/src/base_bigstring_stubs.c
+++ b/src/base_bigstring_stubs.c
@@ -29,14 +29,10 @@
 #define bswap_16 swap16
 #define bswap_32 swap32
 #define bswap_64 swap64
-#elif __CYGWIN__
-#include <endian.h>
 #else
 #include <sys/types.h>
 #if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <sys/endian.h>
-#else
-#include <endian.h>
 #endif
 #define __BYTE_ORDER    _BYTE_ORDER
 #define __LITTLE_ENDIAN _LITTLE_ENDIAN
