--- ./configure.in
+++ ./configure.in
@@ -113,16 +113,6 @@
 # ocamlfind
 AC_CHECK_PROG(USEOCAMLFIND,ocamlfind,yes,no)
 
-if test "$USEOCAMLFIND" = yes; then
-   OCAMLFINDLIB=$(ocamlfind printconf stdlib)
-   OCAMLFIND=$(which ocamlfind)
-   if test "$OCAMLFINDLIB" != "$OCAMLLIB"; then
-   USEOCAMLFIND=no;
-   echo "but your ocamlfind is not compatible with your ocamlc:"
-   echo "ocamlfind : $OCAMLFINDLIB, ocamlc : $OCAMLLIB"
-   fi
-fi
-
 if test "$LIBDIR" = ""; then
    if test "$USEOCAMLFIND" = yes; then
         LIBDIR=$(ocamlfind printconf destdir)/hashcons
