--- ./src/topkg_os.ml
+++ ./src/topkg_os.ml
@@ -213,14 +213,14 @@ module Cmd = struct
     let redirect fd f = strf " %d>%s" fd (Filename.quote f) in
     let stdout = match stdout with None -> "" | Some f -> redirect 1 f in
     let stderr = match stderr with None -> "" | Some f -> redirect 2 f in
-    let win_quote = if Sys.win32 then "\"" else "" in
+    let win_quote = "" in
     strf "%s%s%s%s%s" win_quote cmd stdout stderr win_quote
 
   let exec ?stdout ?stderr cmd =
     try
       let line = line ?stdout ?stderr cmd in
       Topkg_log.debug (fun m -> m ~header:"EXEC" "@[<1>[%s]@]" line);
-      Ok ((), (cmd, `Exited (Sys.command line)))
+      Ok ((), (cmd, `Exited (Sys.command (Printf.sprintf "bash -c '%s'" line))))
     with Sys_error e | Failure e -> R.error_msg e
 
   (* Command existence *)

