--- ./myocamlbuild.ml
+++ ./myocamlbuild.ml
@@ -54,7 +54,7 @@
     flag ["c"; "compile"] & S[A"-ccopt"; A"-I/usr/local/include"; A"-I"; A sodium_include_dir];
     flag ["c"; "ocamlmklib"] & S[A"-L/usr/local/lib"; A("-L"); A sodium_libdir];
     flag ["ocaml"; "link"; "native"; "program"] &
-      S[A"-cclib"; A"-L/usr/local/lib"];
+      S[A"-cclib"; A"-L/usr/local/lib"; A("-L"); A sodium_libdir];
 
     (* Linking cstubs *)
     flag ["c"; "compile"; "use_ctypes"] & S[A"-I"; A ctypes_libdir];
@@ -63,7 +63,7 @@
     (* Linking sodium *)
     flag ["c"; "compile"; "use_sodium"] &
       S[A"-ccopt"; A"--std=c99 -Wall -pedantic -Werror -Wno-pointer-sign"];
-    flag ["c"; "ocamlmklib"; "use_sodium"] & A"-lsodium";
+    flag ["c"; "ocamlmklib"; "use_sodium"] & S[A("-L"); A sodium_libdir; A"-lsodium"];
 
     (* Linking generated stubs *)
     dep ["ocaml"; "link"; "byte"; "library"; "use_sodium_stubs"]
@@ -74,7 +74,7 @@
     dep ["ocaml"; "link"; "native"; "library"; "use_sodium_stubs"]
       ["lib/libsodium_stubs"-.-(!Options.ext_lib)];
     flag ["ocaml"; "link"; "native"; "library"; "use_sodium_stubs"] &
-      S[A"-cclib"; A"-lsodium_stubs"; A"-cclib"; A"-lsodium"];
+      S[A"-cclib"; A"-lsodium_stubs"; A"-cclib"; A("-L"); A"-cclib"; A sodium_libdir; A"-cclib"; A"-lsodium"];
 
     (* Linking tests *)
     flag ["ocaml"; "link"; "byte"; "program"; "use_sodium_stubs"] &
@@ -82,7 +82,7 @@
     dep ["ocaml"; "link"; "native"; "program"; "use_sodium_stubs"]
       ["lib/libsodium_stubs"-.-(!Options.ext_lib)];
     flag ["ocaml"; "link"; "native"; "program"; "use_sodium_stubs"] &
-      S[A"-cclib"; A"-lsodium"];
+      S[ A"-cclib"; A("-L"); A"-cclib"; A sodium_libdir; A"-cclib"; A"-lsodium"];
 
   | _ -> ()
 end;;

