--- ./src/dune
+++ ./src/dune
@@ -1,6 +1,6 @@
 (rule (targets config.h thread_id.h) (deps)
  (action
-  (bash "cp %{lib:jst-config:config.h} %{lib:jst-config:thread_id.h} .")))
+  (bash "cp '%{lib:jst-config:config.h}' '%{lib:jst-config:thread_id.h}' .")))
 
 (library (name core) (public_name core)
  (libraries core_kernel.caml_unix core_kernel core_kernel.base_for_tests
--- ./bigstring_unix/src/dune
+++ ./bigstring_unix/src/dune
@@ -3,7 +3,7 @@
  (preprocessor_deps config.h) (preprocess (pps ppx_jane)))
 
 (rule (targets config.h) (deps)
- (action (bash "cp %{lib:jst-config:config.h} .")))
+ (action (bash "cp '%{lib:jst-config:config.h}' .")))
 
 (rule (targets socketaddr.h) (deps)
- (action (bash "cp %{lib:core:socketaddr.h} .")))
\ No newline at end of file
+ (action (bash "cp '%{lib:core:socketaddr.h}' .")))
\ No newline at end of file
--- ./iobuf_unix/src/dune
+++ ./iobuf_unix/src/dune
@@ -4,4 +4,4 @@
  (preprocess (pps ppx_jane)))
 
 (rule (targets config.h) (deps)
- (action (bash "cp %{lib:jst-config:config.h} .")))
\ No newline at end of file
+ (action (bash "cp '%{lib:jst-config:config.h}' .")))
\ No newline at end of file
--- ./linux_ext/src/dune
+++ ./linux_ext/src/dune
@@ -3,4 +3,4 @@
  (preprocessor_deps config.h) (preprocess (pps ppx_jane)))
 
 (rule (targets config.h) (deps)
- (action (bash "cp %{lib:jst-config:config.h} .")))
\ No newline at end of file
+ (action (bash "cp '%{lib:jst-config:config.h}' .")))
\ No newline at end of file
--- ./time_stamp_counter/src/dune
+++ ./time_stamp_counter/src/dune
@@ -4,4 +4,4 @@
  (libraries core))
 
 (rule (targets config.h) (deps)
- (action (bash "cp %{lib:jst-config:config.h} %{targets}")))
\ No newline at end of file
+ (action (bash "cp '%{lib:jst-config:config.h}' %{targets}")))
\ No newline at end of file
--- ./syslog/src/dune
+++ ./syslog/src/dune
@@ -1,2 +1,2 @@
 (library (name syslog) (public_name core.syslog) (libraries core)
- (c_names syslog_stubs) (preprocess (pps ppx_jane)))
\ No newline at end of file
+ (c_flags (:standard -I/usr/include)) (c_names syslog_stubs) (preprocess (pps ppx_jane)))
\ No newline at end of file
