--- ./test/irmin-pack/cli/dune
+++ ./test/irmin-pack/cli/dune
@@ -1,9 +1,9 @@
-(executable
- (name irmin_fsck)
- (public_name irmin_fsck)
- (package irmin-pack)
- (modules irmin_fsck)
- (libraries irmin-pack irmin-pack.layered))
+; (executable
+;  (name irmin_fsck)
+;  (public_name irmin_fsck)
+;  (package irmin-pack)
+;  (modules irmin_fsck)
+;  (libraries irmin-pack irmin-pack.layered))
 
 (executable
  (name generate)
